controllers/csiaddonsnode_controller.go

```diff
@@ -39,17 +50,98 @@ type CSIAddonsNodeReconciler struct {
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the CSIAddonsNode object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
 func (r *CSIAddonsNodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx)
+
+	logger.Info("Starting Reconcile")
+	// Fetch CSIAddonsNode instance
+	instance := &csiaddonsv1alpha1.CSIAddonsNode{}
+	err := r.Client.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			logger.Info("CSIAddonsNode resource not found")
+
```

Comment:
if the object doesn't exist check if the connection exists in the pool and remove it?

Reply:
> if the object doesn't exist check if the connection exists in the pool and remove it?

The object cannot get deleted until finalizer is removed, so we should be fine here.